front-end/src/components/CreatePost.tsx

```diff
@@ -4,26 +4,37 @@ import { Label, Modal, ModalBody, ModalFooter, ModalHeader } from 'reactstrap';
 import { Post } from '../types/Post';
 import { Form, Input } from 'reactstrap';
 
-export default function CreatePostComponent() {
-    
+const CreatePostComponent = (props: any) => {
+
     const [title, setTitle] = useState('');
     const [desc, setDesc] = useState('');
     const [contentType, setContentType] = useState('');
     const [content, setContent] = useState('');
     const [categories, setCategories] = useState('');
-    const [visibility, setVisibility] = useState('');
-    const [unlisted, setUnlisted] = useState('');
+    const [visibility, setVisibility] = useState('Public');
+    const [unlisted, setUnlisted] = useState(false);
+
+    function changeVisibility(isChecked: boolean) {
+        if (isChecked) {
+            setVisibility("Friends")
+        }
+        else {
+            setVisibility("Public")
+        }
+    }
 
     function sendPost(e:any) {
-        axios.post(process.env.REACT_APP_API_URL + "/api/author/username/posts/", {
+        let data = {
             title: title,
             description: desc,
             visibility: visibility,
             unlisted: unlisted,
             contentType: contentType,
             content: content,
-            categories: categories
-        })
+            categories: [categories]
```

Comment:
We might need to create a new issue about how to go about this. Currently if I type `'images, pets'` into the categories field, it will be sent to the server as 
```javascript
 ['images, pets']
``` 
rather than 
```javascript
['images','pets']
```
That being said categories doesn't seem to be a big deal in the user stories. What do you think about this?

Reply:
I've changed this up for now to be split by whitespace. I think for now its okay, but when this gets changed to a modal this can be more fully fleshed out. I was thinking for this it could use something like adding tags: https://www.npmjs.com/package/react-tag-input

Reply:
Good idea! As you've said, that would be pretty nice to have later on